src/java/org/apache/cassandra/io/sstable/SSTableHeaderFix.java

```diff
@@ -301,7 +301,7 @@ void processFileOrDirectory(Path path)
               .filter(p -> {
                   try
                   {
-                      return Descriptor.fromFilenameWithComponent(new File(p.toFile())).right.type == Component.Type.DATA;
+                      return Descriptor.fromFilenameWithComponent(new File(p.toFile())).right.type == Component.Type.DATA; // checkstyle: permit this invocation
```

Comment:
do we need to allow? `org.apache.cassandra.io.util.File#File(java.nio.file.Path)` we can just pass in the `p`?  this compiles for me locally

```
return Descriptor.fromFilenameWithComponent(new File(p)).right.type == Component.Type.DATA;
```

Reply:
That looks good. `SSTableHeaderFixTest` passes just fine w/ `new File(p)`.